File: javascript/bob/bob.js
export const hey = (message) => {
    if (/^[^a-z]*[A-Z][^a-z]*\? *$/.test(message)) return "Calm down, I know what I'm doing!"
    if (/^[^a-z]*[A-Z][^a-z]*$/.test(message)) return 'Whoa, chill out!'
    if (/\?\s*$/.test(message)) return "Sure."
    if (/^\s*$/.test(message)) return "Fine. Be that way!"
    return "Whatever."
};

File: go/bob/bob.go
// Package bob implements a solution for the bob exercise from the Exercism Go track.
package bob

// Hey takes a message as input, and returns Bob's response to that remark.
func Hey(message string) string {
	remark := newRemark(message)

	switch {
	case remark.isSilence():
		return "Fine. Be that way!"
	case remark.isExasperated():
		return "Calm down, I know what I'm doing!"
	case remark.isShouting():
		return "Whoa, chill out!"
	case remark.isQuestion():
		return "Sure."
	default:
		return "Whatever."
	}
}
